core/chains/evm/client/simulated_backend_client.go

```diff
@@ -470,13 +473,10 @@ func (c *SimulatedBackendClient) BatchCallContext(ctx context.Context, b []rpc.B
 			if !is {
 				return errors.Errorf("SimulatedBackendClient expected first arg to be a string for eth_getBlockByNumber, got: %T", elem.Args[0])
 			}
-			isFullTx, is := elem.Args[1].(bool)
+			_, is = elem.Args[1].(bool)
 			if !is {
 				return errors.Errorf("SimulatedBackendClient expected second arg to be a boolean for eth_getBlockByNumber, got: %T", elem.Args[1])
 			}
-			if isFullTx {
-				return errors.New("SimulatedBackendClient doesn't support full transactions for eth_getBlockByNumber")
-			}
```

Comment:
I thought this would be OK to leave this in, since using `*evmtypes.Block` with `isFullTx==false` is a valid thing to do normally. Was it actively causing a problem?

Reply:
I was thinking that returning an error for that unsupported case could be a _feature_, since without an error the dev could be misled to believe there are just empty blocks without txs :shrug: I don't know the surrounding context well though

Reply:
I understand, but this is a test utility.
If a dev uses that feature for a test, but then makes incorrect assumptions, then they are just writing an invalid test.
Assuming that they got empty blocks without Tx is their fault, as this is a test class, and they should be setting this to respond with the right Txs that they expect.

Reply:
But why set them up for failure in the first place? This code does not support that call. Why is it better to return the wrong answer than to refuse to answer?

Reply:
> I understand, but this is a test utility.

Also, I fundamentally disagree with having lower standards for test utilities. It is a burden on development and undermines our ability to actually confirm production quality with the tests.

Reply:
> I understand, but this is a test utility.

Also, I fundamentally disagree with having lower standards for test utilities. It is a burden on development and undermines our ability to actually confirm production quality with the tests.